src/useCases/request.useCases.ts

```diff
@@ -39,55 +39,139 @@ export default class implements IRequestUseCases {
 
         const { applicant, approvalsNeeded } = requestDetails;
-        const requestProps = { type: REQUEST_TYPE.ADD_DIS_GROUP, applicant, approvalsNeeded };
+        const requestProps = { type: REQUEST_TYPE.ADD_DIS, applicant, approvalsNeeded };
         const payload = { groupId: requestDetails.groupId, disUniqueId: requestDetails.disUniqueId };
 
         const requestNumber = (await this.requestRepo.count()) + 1;
 
         const newRequest: Request = Request.createNew({ ...requestProps, payload, requestNumber });
 
-        const res = await this.requestRepo.create(newRequest, REQUEST_TYPE.ADD_DIS_GROUP);
+        const res = await this.requestRepo.create(newRequest, REQUEST_TYPE.ADD_DIS);
 
         if (!res) throw new InternalError(`Error Creating Add Dis To Group Request: ${payload.groupId} -> ${payload.disUniqueId}`);
 
         return requestNumber;
     };
 
-    public removeDisFromGroup = async (requestDetails: addDisToGroupDTO): Promise<number> => {
-
+    public removeDis = async (requestDetails: disToGroupDTO): Promise<number> => {
         const existsGroup = await this.groupRepo.findById(requestDetails.groupId);
 
         if (!existsGroup) throw new BadRequestError(`Remove dis to non exists group with id ${requestDetails.groupId}`);
 
         const { applicant, approvalsNeeded } = requestDetails;
-        const requestProps = { type: REQUEST_TYPE.REMOVE_DIS_GROUP, applicant, approvalsNeeded };
+        const requestProps = { type: REQUEST_TYPE.REMOVE_DIS, applicant, approvalsNeeded };
         const payload = { groupId: requestDetails.groupId, disUniqueId: requestDetails.disUniqueId };
 
         const requestNumber = (await this.requestRepo.count()) + 1;
 
         const newRequest: Request = Request.createNew({ ...requestProps, payload, requestNumber });
 
-        const res = await this.requestRepo.create(newRequest, REQUEST_TYPE.REMOVE_DIS_GROUP);
+        const res = await this.requestRepo.create(newRequest, REQUEST_TYPE.REMOVE_DIS);
 
-        if (!res) throw new InternalError(`Error Creating Add Dis To Group Request: ${payload.groupId} -> ${payload.disUniqueId}`);
+        if (!res) throw new InternalError(`Error Creating Remove Dis To Group Request: ${payload.groupId} -> ${payload.disUniqueId}`);
+
+        return requestNumber;
+    };
+
+    public addEntities = async (requestDetails: entitiesDTO): Promise<number> => {
+        const existsGroup = await this.groupRepo.findById(requestDetails.groupId);
+
+        if (!existsGroup) throw new BadRequestError(`Add entities to non exists group with id ${requestDetails.groupId}`);
+
+        const { applicant, approvalsNeeded } = requestDetails;
+        const requestProps = { type: REQUEST_TYPE.ADD_ENTITIES, applicant, approvalsNeeded };
+        const payload = { groupId: requestDetails.groupId, entitiesId: requestDetails.entitiesId };
+
+        const requestNumber = (await this.requestRepo.count()) + 1;
```

Comment:
can make conflicts with race conditions, we should return conflict when requestNumber is duplicate key error

Reply:
Why shouldn't try again to set the requestNumber ?